persistence/_index.md

```diff
@@ -2,7 +2,424 @@
+3. Create the `CriteriaQuery` root by calling the `from()` method of the `CriteriaQuery` instance, and passing the entity class for the desired root
+4. Specify the type that the query will return by calling upon the `select()` method of the `CriteriaQuery` instance
+5. Create a `TypedQuery<T>` instance to prepare the query for execution, and specify the type of the query result
+6. Call the `getResultList()` method on the `TypedQuery` instance to execute the query and obtain the results
+
+The process to obtain all objects from the PetOwner entity class is as follows:
+
+```Java
+CriteriaBuilder cb = em.getCriteriaBuilder();
+CriteriaQuery<Pet> cq = cb.createQuery(PetOwner.class);
+Root<PetOwner> petOwner = cq.from(PetOwner.class);
+cq.select(PetOwner);
+TypedQuery<PetOwner> tq = em.createQuery(cq);
+List<PetOwner> allOwners = q.getResultList();
+```
+
+To obtain a selection of attributes from an object, the `Metamodel` API can be used to specify the desired attributes by name. The following code demonstrates how to obtain the lastName attribute of a `PetOwner` object:
+
+```Java
+Root<PetOwner> petOwner = cq.from(PetOwner.class);
+cq.select(petOwner.get(PetOwner_.lastName));
+```
+
+The `where()` method of the `CriteriaQuery` can be used to apply conditions that will filter the query results, much like those provided in JPQL. Again, the individual attributes can be applied to expressions using the metamodel API.
+The Criteria Query API also provides a number of conditional methods to use along with expressions: `equal`, `notEqual`, `lt`, `le`, `gt`, `ge`, `eq`, `between`, and `like`.
+The following code demonstrates how to obtain all `PetOwner` objects where `lastName` is not equal to `'SMITH'`:
+
+```Java
+cq.where(cb.notEqual(petOwner.get(PetOwner_.lastName), "SMITH"));
+```
+
+There are a myriad of other functionalities available with the Criteria Query API. Those include grouping, order by, and the ability to apply functions.
+The following functions are new in Jakarta EE 10: `ceiling()`, `exp()`, `floor()`, `localDate()`, `localDateTime()`, `localTime()`, `ln()`, `power()`, `round()`, and `sign()`.
+Also is the ability to use expressions as conditions within Criteria CASE expressions, new with Jakarta EE 10.
+
+### Named Queries
+
+Named Queries provide a way to create a statically typed query using JPQL and assign it to a name. The name can then be specified to invoke the query.
+Named queries can be specified within an entity class using the `@NamedQueries` annotation. The `@NamedQueries` annotation contains a grouping of `@NamedQuery` annotations, which allow a name to be assigned to a string-based JPQL query.
+The `@NamedNativeQuery` annotation provides the ability to specify native SQL within a named query. The following code shows how to create a named query for the `PetOwner` entity class:
+
+```Java
+@NamedQueries({
```

Comment:
Consider rebasing this section on top of `@NamedQuery` - `@NamedQuery` is repeatable, therefore the use of `@NamedQueries` is not required and in the end allows one to write less code.